### PR TITLE
chore: add a docker-compose file for local development

### PR DESCRIPTION
**Pull request by Gandem**
add a docker-compose file with the locally built and compiled agent as well as the datadog-agent communicating with each other to make local development more seamless

### CODE REVIEW

**Review by r1viollet — DISMISSED** [review]
LGTM
Thanks for improving the onboarding experience
We could have a minor discussion on where to put datadog specific files, though I do not mind them in the current form.

**Review by r1viollet — APPROVED** [review]
LGTM
The cpu arch flag should not be needed for long

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+services:
+  agent:
+    build:
+      context: .
+      args:
+        - arch=${ARCH:?error}
+    privileged: true
+    pid: "host"
+    network_mode: "host"
```
Comment (nsavoire) [inline docker-compose.yml:11]: `host` network mode might not be needed

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    volumes:
+      - ~dd/otel-profiling-agent:/agent
+    command: bash -c "sudo mount -t debugfs none /sys/kernel/debug && make && sudo /agent/otel-profiling-agent -tags 'service:${OTEL_PROFILING_AGENT_SERVICE:-otel-profiling-agent-dev};remote_symbols:yes;cpu_arch:${ARCH:?error}' -collection-agent "http://localhost:8200" -reporter-interval ${OTEL_PROFILING_AGENT_REPORTER_INTERVAL:-60s} -samples-per-second 20 -save-cpuprofile"
```
Comment (nsavoire) [inline docker-compose.yml:14]: otel-profiling-agent could target `http://datadog-agent:8126`

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  agent:
+    build:
+      context: .
+      args:
+        - arch=${ARCH:?error}
+    privileged: true
+    pid: "host"
+    network_mode: "host"
+    volumes:
+      - ~dd/otel-profiling-agent:/agent
```
Comment (nsavoire) [inline docker-compose.yml:13]: I was surprised to see `~dd` work here, why not use `./` instead ?

File: docker-compose.yml (hunk 1 above)
Comment (Gandem) [reply docker-compose.yml]: removed, good call!

File: docker-compose.yml (hunk 3 above)
Comment (Gandem) [reply docker-compose.yml]: oh wow good catch, sorry I missed this when looking at the PR, and must have copy-pasted this part from the commands I was running manually (which work locally due to my workspace's setup)

**Review by nsavoire — APPROVED** [review]
LGTM, nice change making local testing easier !

**Pull request merged**